lib/seattleflu/id3c/cli/command/etl/clinical_retrospectives.py

```diff
@@ -275,6 +276,174 @@ def create_questionnaire_response(record: dict,
+        },
+        '181000124108': {
+            'system': 'http://snomed.info/sct',
+            'code': '181000124108',
+            'display': 'Influenza A virus present',
+        },
+        '441345003': {
+            'system': 'http://snomed.info/sct',
+            'code': '441345003',
+            'display': 'Influenza B virus present',
+        },
+        '441278007': {
+            'system': 'http://snomed.info/sct',
+            'code': '441278007',
+            'display': 'Respiratory syncytial virus untyped strain present',
+        },
+        '440925005': {
+            'system': 'http://snomed.info/sct',
+            'code': '440925005',
+            'display': 'Human rhinovirus present',
+        },
+        '440930009': {
+            'system': 'http://snomed.info/sct',
+            'code': '440930009',
+            'display': 'Human adenovirus present',
+        },
+    }
+
+    # While we ingest inconclusive results for PHSKC samples as null, we do not ingest
+    # inconclusive results for UW Retrospective samples. If other samples need their
+    # inconclusive samples ingested, they should be added to this list.
+    #
+    # TODO: Don't really love this method of deciding inconclusive ingestion. Difficult b/c decision
+    #       should be made in function given how it is passed as an argument to a future diagnostic
+    #       result builder function.
+    sites_ingesting_inconclusives = ['phskc']
+    ingest_inconclusives = True if 'site' in record and record['site'].lower() in sites_ingesting_inconclusives else False
+
+    # Create intermediary, mapped clinical results using SNOMED codes.
+    # This is useful in removing duplicate tests (e.g. multiple tests run for
+    # Influenza A)
+    results = mapped_snomed_test_results(record, ingest_inconclusives)
+
+    # Create observation resources for all results in clinical tests
+    diagnostic_results: Any = {}
+
+    for index, finding in enumerate(results):
+        new_observation = observation_resource('clinical')
+        new_observation['id'] = 'result-' + str(index + 1)
+        new_observation['code']['coding'] = [code_map[finding]]
+        new_observation['valueBoolean'] = results[finding]
+        diagnostic_results[finding] = (new_observation)
+
+    return list(diagnostic_results.values()) or None
+
+
+def mapped_snomed_test_results(record: dict, ingesting_inconclusives: bool) -> Dict[str, bool]:
+    """
+    Given a *record*, returns a dict of the mapped SNOMED clinical
+    finding code and the test result.
+    """
+    redcap_to_snomed_map = {
+        'ncvrt':         '871562009',
+        'result_value' : '871562009', # PHSKC UW Virology Test
+        'revfla':        '181000124108',
+        'fluapr':        '181000124108',
+        'revflb':        '441345003',
+        'flubpr':        '441345003',
+        'revrsv':        '441278007',
+        'revrhn':        '440925005',
+        'revadv':        '440930009',
+    }
```

Comment:
Does changing these snomed codes require updating previously ingested data?

Reply:
Yeah this probably warrants some discussion. I could leave the associated `ncvrt` value unchanged, but when this was first implemented a UK version was used so I think this new code is more reflective of the correct snomed mapping. If we were to update it, future/updated ingestions would be the new value while previously ingested values would map to the old one. In that case it probably would make sense to update old data so that all the targets are the same. I think that the data could either be re-ingested or we could do some sort of direct SQL update where we change any records mapped to the old `target_id` (see below) to the new one.

```
select * from warehouse.presence_absence where target_id = 789
```

Reply:
Ah nice find, I'll dig into if anything else needs updating.

Reply:
Bumping revision number on UW Retros ETL to reingest records with the updated Snomed target, per discussion here: https://seattle-flu-study.slack.com/archives/C0173B4S1LY/p1655408195063209?thread_ts=1655402400.117989&cid=C0173B4S1LY